src/Compilers/Core/Portable/Symbols/ISymbolInternal.cs

```diff
@@ -142,5 +144,14 @@ internal interface ISymbolInternal
         /// is to use it on a symbol that is a definition.
         /// </summary>
         Cci.IReference GetCciAdapter();
+
+        /// <summary>
+        /// <see langword="true"/> if this symbol has any location that is within <paramref name="tree"/>. <see
+        /// langword="false"/> otherwise. Can be more efficient than iteration over all the <see
+        /// cref="ISymbol.Locations"/> as it will avoid an unnecessary array allocation.
+        /// </summary>
+        /// <param name="definedWithinSpan">Optional span.  If present, the location of this symbol must be both inside
+        /// this tree and within the span passed in.</param>
+        bool IsDefinedInSourceTree(SyntaxTree tree, TextSpan? definedWithinSpan, CancellationToken cancellationToken = default);
```

Comment:
>bool IsDefinedInSourceTree(SyntaxTree tree, TextSpan? definedWithinSpan, CancellationToken cancellationToken = default); [](http://example.com/codeflow?start=8&length=120)

I am curious if we should make this API public, i.e. move it to ISymbol.

Reply:
i wouldn't be opposed to that.  it will be interesting to see if anything from the ide layer shows up in a trace (i haven't seen that so far, but there's a lot of other fruit i'm picking currently).